pkgs/applications/editors/rstudio/default.nix

```diff
@@ -1,115 +1,183 @@
+  };
+
+  rsconnectSrc = fetchFromGitHub {
+    owner = "rstudio";
+    repo = "rsconnect";
+    rev = "f5854bb71464f6e3017da9855f058fe3d5b32efd";
+    sha256 = "sha256-ULyWdSgGPSAwMt0t4QPuzeUE6Bo6IJh+5BMgW1bFN+Y=";
+  };
 in
 mkDerivation rec {
   pname = "RStudio";
-  inherit version;
-
-  nativeBuildInputs = [ cmake unzip ant jdk makeWrapper pandoc ];
-
-  buildInputs = [ boost zlib openssl R qtbase qtxmlpatterns qtsensors
-                  qtwebengine qtwebchannel libuuid ];
+  RSTUDIO_VERSION_MAJOR = "1";
+  RSTUDIO_VERSION_MINOR = "4";
+  RSTUDIO_VERSION_PATCH = "1717";
+  version = "${RSTUDIO_VERSION_MAJOR}.${RSTUDIO_VERSION_MINOR}.${RSTUDIO_VERSION_PATCH}";
 
   src = fetchFromGitHub {
     owner = "rstudio";
     repo = "rstudio";
     rev = "v${version}";
-    sha256 = "1n67fa357v51j3z1ma8v2ydfsx3y8n10k2svmfcf4mdzsi8w0kc5";
+    sha256 = "sha256-9c1bNsf8kJjpcZ2cMV/pPNtXQkFOntX29a1cdnXpllE=";
   };
 
+  nativeBuildInputs = [
+    cmake
+    unzip
+    ant
+    jdk
+    makeWrapper
+    pandoc
+    nodejs
+  ];
+
+  buildInputs = [
+    boost
+    zlib
+    openssl
+    R
+    qtbase
+    qtxmlpatterns
+    qtsensors
+    qtwebengine
+    qtwebchannel
+    libuuid
+    libyamlcpp
+    soci
+    postgresql
+  ];
+
+  cmakeFlags = [
+    "-DRSTUDIO_TARGET=Desktop"
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DQT_QMAKE_EXECUTABLE=${qmake}/bin/qmake"
+    "-DRSTUDIO_USE_SYSTEM_SOCI=ON"
+    "-DRSTUDIO_USE_SYSTEM_BOOST=ON"
+    "-DRSTUDIO_USE_SYSTEM_YAML_CPP=ON"
+    "-DPANDOC_VERSION=${pandoc.version}"
+    "-DCMAKE_INSTALL_PREFIX=${placeholder "out"}/lib/rstudio"
+  ];
+
   # Hack RStudio to only use the input R and provided libclang.
-  patches = [ ./r-location.patch ./clang-location.patch
-              (fetchpatch {
-                # Fetch a patch to ensure Rstudio compiles against R
-                # 4.0.0, should be removed next 1.2.X Rstudio update
-                # or possibly 1.3.X
-                url = "https://github.com/rstudio/rstudio/commit/3fb2397c2f208bb8ace0bbaf269481ccb96b5b20.patch";
-                sha256 = "0qpgjy6aash0fc0xbns42cwpj3nsw49nkbzwyq8az01xwg81g0f3";
-              })
-            ];
+  patches = [
+    ./r-location.patch
+    ./clang-location.patch
+    # postFetch doesn't work with this | error: unexpected end-of-file
+    # replacing /usr/bin/node is done in postPatch
+    # https://src.fedoraproject.org/rpms/rstudio/tree/rawhide
+    (fetchpatch {
+      name = "system-node.patch";
+      url = "https://src.fedoraproject.org/rpms/rstudio/raw/5bda2e290c9e72305582f2011040938d3e356906/f/0004-use-system-node.patch";
+      sha256 = "sha256-P1Y07RB/ceFNa749nyBUWSE41eiiZgt43zVcmahvfZM=";
+    })
+  ];
+
   postPatch = ''
     substituteInPlace src/cpp/core/r_util/REnvironmentPosix.cpp --replace '@R@' ${R}
+
+    substituteInPlace src/cpp/CMakeLists.txt \
+      --replace 'SOCI_LIBRARY_DIR "/usr/lib"' 'SOCI_LIBRARY_DIR "${soci}/lib"'
+
+    substituteInPlace src/gwt/build.xml \
```

Comment:
I think this substituteInPlace src/gwt/build.xml is unnecessary if you use Fedora's system-node.patch as above

Reply:
the patch has `/usr/bin/node`